server/android/src/android.rs

```diff
@@ -2,167 +2,89 @@
-
         let cache_dir: String = env.get_string(cache_dir).unwrap().into();
 
-        let cert_path = files_dir.join("certs");
-        if !cert_path.join(PRIVATE_CERT_FILE).exists() {
-            generate_certs(&cert_path);
-        }
+        let settings = Settings {
+            server: ServerSettings {
+                port,
+                danger_allow_http: false,
+                debug_no_access_control: false,
+                cors_origins: vec!["http://localhost".to_string()],
+                base_dir: Some(files_dir.to_str().unwrap().to_string()),
+                machine_uid: Some(android_id),
+            },
+            database: DatabaseSettings {
+                username: "n/a".to_string(),
+                password: "n/a".to_string(),
+                port: 0,
+                host: "n/a".to_string(),
+                database_name: db_path.to_string_lossy().to_string(),
+                // See https://github.com/openmsupply/remote-server/issues/1076
+                init_sql: Some(format!("PRAGMA temp_store_directory = '{}';", cache_dir)),
+            },
+            // sync settings need to be configured at runtime
+            sync: None,
+            logging: None,
+        };
 
         // run server in background thread
         let thread = thread::spawn(move || {
-            actix_web::rt::System::new().block_on(async move {
-                let settings = Settings {
-                    server: ServerSettings {
-                        port,
-                        danger_allow_http: true,
-                        debug_no_access_control: false,
-                        cors_origins: vec!["http://localhost".to_string()],
-                        base_dir: Some(files_dir.to_str().unwrap().to_string()),
-                        machine_uid: Some(android_id),
-                    },
-                    database: DatabaseSettings {
-                        username: "n/a".to_string(),
-                        password: "n/a".to_string(),
-                        port: 0,
-                        host: "n/a".to_string(),
-                        database_name: db_path.to_string_lossy().to_string(),
-                        // See https://github.com/openmsupply/remote-server/issues/1076
-                        init_sql: Some(format!("PRAGMA temp_store_directory = '{}';", cache_dir)),
-                    },
-                    // sync settings need to be configured at runtime
-                    sync: None,
-                    logging: None,
-                };
-                let _ = start_server(settings, off_switch_receiver).await;
-            });
+            actix_web::rt::System::new()
+                .block_on(start_server(settings, off_switch_receiver))
+                .unwrap();
         });
 
         let mut bucket = SERVER_BUCKET.lock().unwrap();
-        bucket.insert(RunningServerHandle { thread, off_switch })
+        *bucket = Some(ServerBucket { off_switch, thread });
```

Comment:
Please check if there is a running server. If you call this method twice you loose the handle and there is no way to join the server thread... While we are not calling it twice we should check here anyway.

Reply:
There was a quite a few comment about this, I'll have a look sometime, but I didn't have any issues with all of the testing and developing

Reply:
Its mainly good practice :) just reduce the amount of required extra knowledge about how the method is called. The reader than also doesn't has to consider error cases that are a no opt, e.g. that Android doesn't call onStart multiple times, the code is correct in any case. Also wouldn't complain as much if it wouldn't have been done cleanly before.